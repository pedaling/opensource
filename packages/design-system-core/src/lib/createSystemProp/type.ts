import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
