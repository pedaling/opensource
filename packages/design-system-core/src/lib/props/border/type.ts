import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
