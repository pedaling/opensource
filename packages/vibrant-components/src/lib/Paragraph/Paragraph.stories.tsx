import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Paragraph } from './Paragraph';

export default {
  title: 'Paragraph',
  component: Paragraph,
  args: {
    level: 1,
    children: 'Sample Text',
  },
} as ComponentMeta<typeof Paragraph>;

export const Basic: ComponentStory<typeof Paragraph> = ({ children, ...restProps }) => (
  <Paragraph {...restProps}>{children}</Paragraph>
);
