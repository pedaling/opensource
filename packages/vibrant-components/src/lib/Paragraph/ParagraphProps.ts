import type { ResponsiveValue } from '@vibrant-ui/core';
import { propVariant, withVariation } from '@vibrant-ui/core';
import type { TextProps } from '../Text';

export type ParagraphProps = Omit<TextProps, 'kind'> & {
  level: ResponsiveValue<1 | 2 | 3>;
};

export const withParagraphVariation = withVariation<ParagraphProps>()(
  propVariant({
    props: [
      {
        name: 'level',
        responsive: true,
      },
    ],
    variants: ({ level }) => ({
      kind: `paragraph${level}` as const,
    }),
  })
);
