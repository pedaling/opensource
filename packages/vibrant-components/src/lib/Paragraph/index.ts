export { Paragraph } from './Paragraph';
export type { ParagraphProps } from './ParagraphProps';
