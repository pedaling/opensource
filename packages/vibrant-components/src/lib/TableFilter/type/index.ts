export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
