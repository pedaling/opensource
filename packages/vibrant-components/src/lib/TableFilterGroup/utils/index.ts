import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
