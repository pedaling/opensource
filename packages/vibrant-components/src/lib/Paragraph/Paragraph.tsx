import { Text } from '@vibrant-ui/core';
import { withParagraphVariation } from './ParagraphProps';

export const Paragraph = withParagraphVariation(({ innerRef, ...restProps }) => <Text ref={innerRef} {...restProps} />);
