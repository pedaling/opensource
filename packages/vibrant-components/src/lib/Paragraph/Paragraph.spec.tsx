import { baseTheme } from '@vibrant-ui/theme';
import type { ReactRenderer } from '@vibrant-ui/utils/testing';
import { createReactRenderer } from '@vibrant-ui/utils/testing';
import { Paragraph } from './Paragraph';

describe('<Paragraph />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  let element: HTMLElement;

  describe('when Paragraph with level 3 rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <Paragraph data-testid="paragraph" level={3}>
          Hello World
        </Paragraph>
      );

      element = await renderer.findByTestId('paragraph');
    });

    it('paragraph3 style applied', () => {
      const themeTypography = baseTheme.typography;
      const themeTypographyWeight = baseTheme.typographyWeight;

      expect(element).toHaveStyleRule('font-size', themeTypography.paragraph3.fontSize);

      expect(element).toHaveStyleRule(
        'font-weight',
        themeTypographyWeight[themeTypography.paragraph3.fontWeight].fontWeight.toString()
      );

      expect(element).toHaveStyleRule('line-height', themeTypography.paragraph3.lineHeight);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
