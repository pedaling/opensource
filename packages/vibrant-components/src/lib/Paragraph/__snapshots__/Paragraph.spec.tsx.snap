// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> when Paragraph with level 3 rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-family: Pretendard;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.75rem;
  color: #101010;
}

<div>
  <span
    class="emotion-0"
    data-testid="paragraph"
  >
    Hello World
  </span>
</div>
`;
